### PR TITLE
Switch to Apache 2.0 license

### PR DESCRIPTION
**Pull request by fritzo**
Addresses #2249 #2067 

1. Switch from MIT to Apache-2.0 license
    - update LICENSE.md
    - update setup.py
2. Add a script to add copyright notices and SPDX identifiers to all .py and .cpp files.
3. Apply the script to all source files

I plan to update and modify the script as we e.g. update the copyright year next year or add new file types.

After this PR merges we can disable the Uber CLA 🎉 

@neerajprad we will also need to switch licenses in other repos:
- https://github.com/pyro-ppl/numpyro
- https://github.com/pyro-ppl/funsor
- https://github.com/pyro-ppl/pyro-api

### CODE REVIEW

**Comment by fritzo** [comment]
cc @swinslow @ibrahimhaddad

File: scripts/update_headers.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# Copyright (c) 2020 Contributors to the Pyro project.
```
Comment (fritzo) [inline scripts/update_headers.py]: @neerajprad here is the new script file

Comment (neerajprad) [reply scripts/update_headers.py]: Thanks, @fritzo. Should we do this for the other projects in pyro-ppl too?

Comment (fritzo) [reply scripts/update_headers.py]: Yes, we should do this for all repos in https://github.com/pyro-ppl.

Comment (neerajprad) [reply scripts/update_headers.py]: Okay, I'll make the changes for numpyro and pyro-api.

**Comment by fritzo** [comment]
@neerajprad I've updated to use @swinslow's recommended simpler copyright notices.

**Comment by fritzo** [comment]
@neerajprad I reverted to preserve Uber copyrights on existing files; the `scripts/update_headers.py` can remain as-is, since we can use @swinslow's simplified copyright notice for new files.

**Pull request merged**